### DISCUSSION TITLE
Looking for an AI-Assisted Tool to Easily Create WordPress Plugins as a Non-Developer

### DISCUSSION THREAD

**Discussion by ghost** [community/community#143569] [Programming Help]
### Select Topic Area

Question

### Body

I’m looking for an easy way to create WordPress plugins with guidance and support from artificial intelligence, specifically a tool that’s user-friendly for someone who isn’t a developer. I’ve tried using AI assistants like ChatGPT and Claude AI to create basic plugins, but these methods aren’t very practical for more complex projects. I also tested Cofounder, but it seems to focus on building full-scale software applications, which exceeds my needs.

I’m seeking a tool or a GitHub project specifically designed for WordPress plugin development that offers intuitive, step-by-step guidance suitable for non-technical users.

Thank you for your assistance!

**Comment by LiteBrite82** [comment]
GitHub copilot or [GitHub Spark](https://githubnext.com/projects/github-spark) could be helpful options.

**Comment by michaelaltao** [comment]
Hi @Benspy22 
Hope you are doing well.

For non-technical users interested in WordPress plugin development, one great choice is WP CLI Plugin Scaffolding or Plugin Boilerplate projects available on GitHub. These tools simplify the process and make plugin creation more approachable by offering structured, step-by-step scaffolding.

Recommended Options:

WP CLI Plugin Scaffolding:
Repository: https://github.com/wp-cli/scaffold-command
Features: WP-CLI is a command-line tool for managing WordPress installations, and this plugin scaffolding command can help non-technical users create plugins by simply running a few commands. This project sets up the basic structure and files you need for a plugin, which can then be modified as needed.
Guidance: It provides extensive documentation and support for users to get started without much technical background, focusing on simplifying the boilerplate setup. WP-CLI also has an intuitive syntax, making it easier for beginners.

WordPress Plugin Boilerplate:
Repository: https://github.com/DevinVinson/WordPress-Plugin-Boilerplate
Features: This boilerplate provides a solid foundation for building a plugin by pre-organizing all files and folders. It offers a well-structured way to include core plugin code, assets, and additional helper classes, making it easy to extend functionality.
Guidance: It includes detailed comments and examples, explaining what each section of the code is meant to do. The step-by-step setup process is well-documented, so even users with limited programming knowledge can follow along and modify as needed.

Hope to be helpful for you!
If then, please mark up this answer as done~~

Thank you

**Comment by queenofcorgis** [comment]
@Benspy22 we're going through our old posts - did any of the above suggestions do the job for you, if so could you please mark the answer or close this out?
